ArduPilot/ardupilot
# AP_DDS: update dds profile, eliminate need for integration service

Use type and service name mapping rules to ensure `AP_DDS` service calls are automatically mapped between DDS and ROS 2. This eliminates the need to run the eProsima Integration Service to access AP_DDS services from the ROS 2 CLI.  

Additionally the `arm_motors` service has been placed in the `ap` namespace to be consistent with other topics.

## Details

- Use type and topic name mapping rules in the arm_motors service profile.
- Remove the integration service configuration file.
- Update the service section in the README and document the topic and service mapping rules.

Figure: arm_motors service request/response pair with types and names in eProsima `fastddsmonitor`. 
![dds-monitor-services](https://github.com/ArduPilot/ardupilot/assets/24916364/7133ee93-31f9-45fe-9804-ded9b183017f)

## Testing

- Configure for SITL with `./waf configure --board sitl --enable-dds`
- Build `./waf build`
- Run a SITL session with DDS enabled and use UDP as a transport layer

```bash
ros2 launch ardupilot_sitl sitl_dds_udp.launch.py \
transport:=udp4 \
refs:=$(ros2 pkg prefix ardupilot_sitl)/share/ardupilot_sitl/config/dds_xrce_profile.xml \
synthetic_clock:=True \
wipe:=False \
model:=quad speedup:=1 \
slave:=0 \
instance:=0 \
defaults:=$(ros2 pkg prefix ardupilot_sitl)/share/ardupilot_sitl/config/default_params/copter.parm,$(ros2 pkg prefix ardupilot_sitl)/share/ardupilot_sitl/config/default_params/dds_udp.parm \
sim_address:=127.0.0.1 \
master:=tcp:127.0.0.1:5760 \
sitl:=127.0.0.1:5501
```

- Check services are available

```bash
$ ros2 service list                        
/ap/arm_motors
```

- Run the service example and check output in mavproxy

```bash
$ ros2 service call /ap/arm_motors ardupilot_msgs/srv/ArmMotors "{arm: True}"
requester: making request: ardupilot_msgs.srv.ArmMotors_Request(arm=True)

response:
ardupilot_msgs.srv.ArmMotors_Response(result=True)
```
